tests/data/SPECS/dynamic.spec

```diff
@@ -21,6 +23,13 @@ echo "Q: Why?\nA: Because we can!" > FAQ
 mkdir -p $RPM_BUILD_ROOT/usr/local/bin
 echo " " > $RPM_BUILD_ROOT/usr/local/bin/hello
 
+%{?FULLDYNAMIC:
+echo "Group: Utilities" >> %{specpartsdir}/docs.specpart
+echo "License: GPL" >> %{specpartsdir}/docs.specpart
+echo "Distribution: RPM test suite." >> %{specpartsdir}/docs.specpart
+echo "URL: http://rpm.org" >> %{specpartsdir}/docs.specpart
+echo "Summary: dynamic hello -- hello, world rpm" >> %{specpartsdir}/docs.specpart
+}
```

Comment:
I'd put these to separate file, just to get more coverage.

Also, do add a separate test for a syntax error while parsing dynamically generated stuff, which is a different case from checking for required tags (for which there is a test). The error message in the syntax case is pretty terrible because there's no file information attached to the message (although there are hints in the output log), but AFAICS that doesn't change here so I'll just file a separate ticket on that.